src/sagemaker/local/local_session.py

```diff
@@ -53,8 +53,8 @@ def __init__(self, sagemaker_session=None):
         """
         self.sagemaker_session = sagemaker_session or LocalSession()
 
-    def create_training_job(self, TrainingJobName, AlgorithmSpecification, InputDataConfig, OutputDataConfig,
-                            ResourceConfig, **kwargs):
+    def create_training_job(self, TrainingJobName, AlgorithmSpecification, OutputDataConfig,
+                            ResourceConfig, InputDataConfig=None, **kwargs):
```

Comment:
if the first line of this method becomes

`InputDataConfig = InputDataConfig or {}`

Then you don't need to actually touch any other part of the code as all the for loops will just not happen.